apps/frontend/src/components/NeighborhoodSearch/NeighborhoodSearch.tsx

```diff
@@ -0,0 +1,172 @@
+import { useEffect, useState } from 'react';
+import { Form, Container, Row, Col } from 'react-bootstrap';
+import { Neighborhood, NeighborhoodsPerPage } from '@neighborhood/backend/src/types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { AxiosError } from 'axios';
+import { faPlus } from '@fortawesome/free-solid-svg-icons';
+import InfiniteScroll from 'react-infinite-scroll-component';
+import styles from './NeighborhoodSearch.module.css';
+import NeighborhoodCard from '../NeighborhoodCard/NeighborhoodCard';
+import CustomBtn from '../CustomBtn/CustomBtn';
+import CreateNeighborhoodModal from '../CreateNeighborhoodModal/CreateNeighborhoodModal';
+import neighborhoodsService from '../../services/neighborhoods';
+import SpinWheel from '../SpinWheel/SpinWheel';
+import AlertBox from '../AlertBox/AlertBox';
+import { ErrorObj } from '../../types';
+
+export default function NeighborhoodSearch({
+  neighborhoods,
+  cursor,
+  isNextPage,
+}: {
+  neighborhoods: Neighborhood[];
+  cursor?: number;
+  isNextPage: boolean;
+}) {
+  const [neighborhoodList, setNeighborhoodList] = useState(neighborhoods);
+  const [hasNextPage, setHasNextPage] = useState(isNextPage);
+  const [currentCursor, setCurrentCursor] = useState(cursor);
```

Comment:
Same here

Reply:
hmm not sure tbh. What's your proposed alternative?

Reply:
I'd propose we use https://react.dev/reference/react/useRef in the future when we need to store data in components that do not need to trigger a re-render.
In this case, I think 'hasNextPage' and 'currentCursor' do not need to trigger a re-render.

Reply:
@radu-constantin I've made the updates. Check it out when you have the time!